### PR TITLE
feat: cli uninstall

### PR DESCRIPTION
**Pull request by anchenyi**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/28553259

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -529,6 +529,30 @@
   "core.copilot.addAPI.success": "%s have(has) been successfully added to %s",
   "core.copilot.addAPI.InjectAPIKeyActionFailed": "Inject API key action to teamsapp.yaml file unsuccessful, make sure the file contains teamsApp/create action in provision section.",
   "core.copilot.addAPI.InjectOAuthActionFailed": "Inject OAuth action to teamsapp.yaml file unsuccessful, make sure the file contains teamsApp/create action in provision section.",
+  "core.uninstall.botNotFound": "Cannot find bot using the manifest ID %s",
+  "core.uninstall.confirm.tdp": "App registration of manifest ID: %s will be uninstalled. Please confirm.",
+  "core.uninstall.confirm.m365App": "M365 App of Title ID: %s will be uninstalled. Please confirm.",
+  "core.uninstall.confirm.bot": "Bot framework registration of bot ID: %s will be uninstalled. Please confirm.",
+  "core.uninstall.confirm.cancel.tdp": "App registration uninstallation cancelled.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:536]: Proposed: Uninstallation of app registration is cancelled.

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -529,6 +529,30 @@
...
+  "core.uninstall.confirm.cancel.m365App": "M365 App uninstallation cancelled.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:537]: Proposed: Uninstallation of M365 App is cancelled.

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -529,6 +529,30 @@
...
+  "core.uninstall.confirm.cancel.bot": "Bot framework registration uninstallation cancelled.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:538]: Proposed: Uninstallation of Bot framework registration is cancelled.

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -529,6 +529,30 @@
...
+  "core.uninstall.success.tdp": "App registration of manifest ID: %s has been successfully uninstalled.",
+  "core.uninstall.success.m365App": "M365 App of Title ID: %s has been successfully uninstalled.",
+  "core.uninstall.success.delayWarning": "The uninstallation of the M365 app might have some delay.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:541]: Proposed: The uninstallation of the M365 app may be delayed.

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -529,6 +529,30 @@
...
+  "core.uninstall.success.bot": "Bot framework registration of bot ID: %s has been successfully uninstalled.",
+  "core.uninstall.failed.titleId": "Failed to find the Title ID. This app is probably not installed.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:543]: Proposed: Unable to find the Title ID. This app is probably not installed.

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: Updated.

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: Updated.

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: Updated.

File: packages/fx-core/resource/package.nls.json (hunk 4 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: Updated.

File: packages/fx-core/resource/package.nls.json (hunk 5 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: Updated. Thanks!

File: packages/cli/src/commands/models/m365Unacquire.ts [hunk 6]
```diff
@@ -1,57 +1,75 @@
...
       description: commands.uninstall.options["manifest-id"],
       type: "string",
     },
+    {
+      name: QuestionNames.Env,
+      description: commands.uninstall.options["env"],
+      type: "string",
+    },
+    {
+      name: QuestionNames.ProjectPath,
```
Comment (jayzhang) [inline packages/cli/src/commands/models/m365Unacquire.ts:38]: suggest to use global option `ProjectFolderOption`

Comment (anchenyi) [reply packages/cli/src/commands/models/m365Unacquire.ts:38]: ProjectFolderOption's "required" property is true, but in cli uninstall, the project path is optional.

Comment (jayzhang) [reply packages/cli/src/commands/models/m365Unacquire.ts:38]: Though `ProjectFolderOption` is required, it has a current folder as default value, this option should be consistant globally for the TTK CLI.

Comment (anchenyi) [reply packages/cli/src/commands/models/m365Unacquire.ts]: Adjusted the naming. Thanks.

**Review by jayzhang — APPROVED** [review]

File: packages/cli/src/resource/commands.json [hunk 7]
```diff
@@ -144,10 +144,14 @@
     "description": "Run the publish stage in teamsapp.yml."
   },
   "uninstall": {
-    "description": "Remove an acquired M365 App.",
+    "description": "Clean up resources associated with Manifest ID, Title ID or an environment in Teams Toolkit generated project. Resources including application registration in Teams Developer Portal, bot registration in Bot Framework Portal and uploaded custom applications in Microsoft 365 apps.",
```
Comment (supkasar) [inline packages/cli/src/resource/commands.json:147]: Proposed: Clean up resources associated with Manifest ID, Title ID, or an environment in Teams Toolkit generated project. Resources include app registration in Teams Developer Portal, bot registration in Bot Framework Portal, and uploaded custom apps in Microsoft 365 apps.

File: packages/fx-core/resource/package.nls.json [hunk 8]
```diff
@@ -529,6 +529,33 @@
   "core.copilot.addAPI.success": "%s have(has) been successfully added to %s",
   "core.copilot.addAPI.InjectAPIKeyActionFailed": "Inject API key action to teamsapp.yaml file unsuccessful, make sure the file contains teamsApp/create action in provision section.",
   "core.copilot.addAPI.InjectOAuthActionFailed": "Inject OAuth action to teamsapp.yaml file unsuccessful, make sure the file contains teamsApp/create action in provision section.",
+  "core.uninstall.botNotFound": "Cannot find bot using the manifest ID %s",
+  "core.uninstall.confirm.tdp": "App registration of manifest ID: %s will be uninstalled. Please confirm.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:533]: Proposed: App registration of manifest ID: %s will be canceled. Please confirm.

File: packages/fx-core/resource/package.nls.json [hunk 9]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstall.confirm.m365App": "Microsoft 365 Application of Title ID: %s will be uninstalled. Please confirm.",
+  "core.uninstall.confirm.bot": "Bot framework registration of bot ID: %s will be uninstalled. Please confirm.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:535]: Proposed: Bot framework registration of bot ID: %s will be canceled. Please confirm.

File: packages/fx-core/resource/package.nls.json (hunk 8 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json:533]: Hi @supkasar, in this feature, we use "uninstall" to represent the resources clean up, so I think using "uninstalled" here is more appropriate.

File: packages/fx-core/resource/package.nls.json [hunk 10]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstall.confirm.cancel.tdp": "Uninstallation of app registration is cancelled.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:536]: Proposed: Removal of app registration is canceled.

File: packages/fx-core/resource/package.nls.json [hunk 11]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstall.confirm.cancel.m365App": "Uninstallation of Microsoft 365 Application is cancelled.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:537]: Proposed: Removal of Microsoft 365 Application is canceled.

File: packages/fx-core/resource/package.nls.json [hunk 12]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstall.confirm.cancel.bot": "Uninstallation of Bot framework registration is cancelled.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:538]: Proposed: Removal of Bot framework registration is canceled.

File: packages/fx-core/resource/package.nls.json [hunk 13]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstall.success.tdp": "App registration of manifest ID: %s has been successfully uninstalled.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:539]: Proposed: App registration of manifest ID: %s successfully removed.

File: packages/fx-core/resource/package.nls.json [hunk 14]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstall.success.m365App": "Microsoft 365 Application of Title ID: %s has been successfully uninstalled.",
+  "core.uninstall.success.delayWarning": "The uninstallation of the Microsoft 365 Application may be delayed.",
+  "core.uninstall.success.bot": "Bot framework registration of bot ID: %s has been successfully uninstalled.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:542]: Proposed: Bot framework registration of bot ID: %s successfully removed.

File: packages/fx-core/resource/package.nls.json [hunk 15]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstall.failed.titleId": "Unable to find the Title ID. This app is probably not installed.",
+  "core.uninstallQuestion.manifestId": "Manifest ID",
+  "core.uninstallQuestion.env": "Environment",
+  "core.uninstallQuestion.titleId": "Title ID",
+  "core.uninstallQuestion.chooseMode": "Choose a way to clean up resources",
+  "core.uninstallQuestion.manifestIdMode": "Manifest ID",
+  "core.uninstallQuestion.manifestIdMode.detail": "Clean up resources associated with Manifest ID. Resources including application registration in Teams Developer Portal, bot registration in Bot Framework Portal and uploaded custom applications in Microsoft 365 apps. The Manifest ID can be found in the environment file (with the default environment key Teams_App_ID) in the project generated by the Teams Toolkit.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:549]: Proposed: Clean up resources associated with Manifest ID. This includes app registration in Teams Developer Portal, bot registration in Bot Framework Portal, and custom apps uploaded to Microsoft 365. You can find the Manifest ID in the environment file (default environment key: Teams_App_ID) in the project created by Teams Toolkit.

File: packages/fx-core/resource/package.nls.json [hunk 16]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstallQuestion.envMode": "Environment in Teams Toolkit Generated Project",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:550]: Proposed: Environment in Teams Toolkit Created Project

File: packages/fx-core/resource/package.nls.json (hunk 10 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json:536]: same as above. "uninstall" shoule be more consistent with the feature.

File: packages/fx-core/resource/package.nls.json [hunk 17]
```diff
@@ -529,6 +529,33 @@
...
+  "core.uninstallQuestion.envMode.detail": "Clean up resources associated with a specific environment in the project created by Teams Toolkit. Resources including application registration in Teams Developer Portal, bot registration in Bot Framework Portal and uploaded custom applications in Microsoft 365 apps.",
+  "core.uninstallQuestion.titleIdMode": "Title ID",
+  "core.uninstallQuestion.titleIdMode.detail": "Uninstall the uploaded custom application associated with Title ID. The Title ID can be found the environment file in the Teams Toolkit generated project.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:553]: Proposed: Uninstall the uploaded custom app associated with Title ID. The Title ID can be found in the environment file in the Teams Toolkit created project.

File: packages/fx-core/resource/package.nls.json (hunk 12 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: same as above. "uninstall" shoule be more consistent with the feature.

File: packages/fx-core/resource/package.nls.json (hunk 8 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: Fixed.

File: packages/fx-core/resource/package.nls.json (hunk 10 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: Fixed

File: packages/fx-core/resource/package.nls.json (hunk 12 above)
Comment (anchenyi) [reply packages/fx-core/resource/package.nls.json]: Fixed

**Review by supkasar — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by kuojianlu — APPROVED** [review]

**Pull request merged**